aeron-driver/src/main/c/media/aeron_udp_channel_transport_fixed_loss.c

```diff
@@ -0,0 +1,267 @@
+
+int aeron_udp_channel_interceptor_fixed_loss_init_incoming(
+    void **interceptor_state, aeron_driver_context_t *context, aeron_udp_channel_transport_affinity_t affinity);
+
+aeron_udp_channel_interceptor_bindings_t *aeron_udp_channel_interceptor_fixed_loss_load(
+    aeron_udp_channel_interceptor_bindings_t *delegate_bindings)
+{
+    aeron_udp_channel_interceptor_bindings_t *interceptor_bindings;
+    if (aeron_alloc((void **)&interceptor_bindings, sizeof(aeron_udp_channel_interceptor_bindings_t)) < 0)
+    {
+        return NULL;
+    }
+
+    interceptor_bindings->incoming_init_func = aeron_udp_channel_interceptor_fixed_loss_init_incoming;
+    interceptor_bindings->outgoing_init_func = NULL;
+    interceptor_bindings->outgoing_send_func = NULL;
+    interceptor_bindings->incoming_func = aeron_udp_channel_interceptor_fixed_loss_incoming;
+    interceptor_bindings->outgoing_close_func = NULL;
+    interceptor_bindings->incoming_close_func = NULL;
+    interceptor_bindings->outgoing_transport_notification_func = NULL;
+    interceptor_bindings->outgoing_publication_notification_func = NULL;
+    interceptor_bindings->outgoing_image_notification_func = NULL;
+    interceptor_bindings->incoming_transport_notification_func = NULL;
+    interceptor_bindings->incoming_publication_notification_func = NULL;
+    interceptor_bindings->incoming_image_notification_func = NULL;
+
+    interceptor_bindings->meta_info.name = "fixed-loss";
+    interceptor_bindings->meta_info.type = "interceptor";
+    interceptor_bindings->meta_info.next_interceptor_bindings = delegate_bindings;
+
+    return interceptor_bindings;
+}
+
+int aeron_udp_channel_interceptor_fixed_loss_configure(const aeron_udp_channel_interceptor_fixed_loss_params_t *fixed_loss_params)
+{
+    aeron_udp_channel_interceptor_fixed_loss_params = fixed_loss_params;
+
+    if (aeron_int64_counter_map_init(
+            &stream_and_session_id_to_offset_map, STREAM_AND_SESSION_ID_NULL_OFFSET, 16, AERON_MAP_DEFAULT_LOAD_FACTOR) < 0)
+    {
+        AERON_SET_ERR(errno, "%s", "Unable to init stream_and_session_id_to_offset_map");
+        return -1;
+    }
+
+    return 0;
+}
+
+void aeron_udp_channel_transport_fixed_loss_load_env(void)
+{
+    aeron_udp_channel_interceptor_fixed_loss_params_t *fixed_loss_params;
+    const char *args = AERON_CONFIG_GETENV_OR_DEFAULT(AERON_UDP_CHANNEL_TRANSPORT_BINDINGS_FIXED_LOSS_ARGS_ENV_VAR, "");
+    char *args_dup = strdup(args);
+    if (NULL == args_dup)
+    {
+        AERON_SET_ERR(errno, "%s", "Duplicating args string");
+        return;
+    }
+
+    if (aeron_alloc((void **)&fixed_loss_params, sizeof(aeron_udp_channel_interceptor_fixed_loss_params_t)) < 0)
+    {
+        return;
+    }
+
+    if (aeron_udp_channel_interceptor_fixed_loss_parse_params(args_dup, fixed_loss_params) >= 0)
+    {
+        // TODO handle return of -1
+        aeron_udp_channel_interceptor_fixed_loss_configure(fixed_loss_params);
+    }
+    else
+    {
```

Comment:
Use `AERON_APPEND_ERR` here to capture additional context.  Also `AERON_APPEND_ERR("%s", "");` is okay to as it helps capture an error trace.